duckduckgo/duckduckgo-autofill
# Bitwarden autofill integration

**Reviewer:** @shakyShane 
**Asana:** https://app.asana.com/0/0/1203088211658380/f
**BSK PR:** https://github.com/duckduckgo/BrowserServicesKit/pull/136
**macOS PR:** https://github.com/more-duckduckgo-org/macos-browser/pull/710

## Description
This adds the Bitwarden-specific UI and API changes. The largest change is a privacy feature that delays reading `availableInputTypes` for credentials until we have actually found a login form. This allows us to decrease the amounts of calls we make to Bitwarden when the app is locked ([Tech Design here](https://app.asana.com/0/481882893211075/1203035091319598/f)).

## Steps to test
Added automated tests. The integration with the macOS platform is still in flight, but you can see the BSK stubs [here](https://github.com/duckduckgo/BrowserServicesKit/pull/136).
